src/scipp/io/hdf5.py

```diff
@@ -225,24 +225,27 @@ def write(group, data):
         # 2 separate groups.
         for view_name, view in zip(['coords', 'masks', 'attrs'], views):
             subgroup = group.create_group(view_name)
-            for name in view:
-                g = VariableIO.write(group=subgroup.create_group(str(name)),
+            for i, name in enumerate(view):
+                var_group_name = f'elem_{i}'
```

Comment:
It is a pity that this makes the HDF5 files less readable/interpretable. Would it make sense to use the actual name unless there are conflicting symbols (but ignore it in the reader code)? Only hurdle is to define a non-clashing alternative name (maybe we can use a prefix for the actual name?).